### PR TITLE
Refactor App Manager Case Search config UI

### PR DESCRIPTION
**Pull request by sravfeyn**
## Summary

Part 1 of https://dimagi-dev.atlassian.net/browse/USH-630

Before

Screenshot 2021-01-28 at 2 17 16 PM

After
Screenshot 2021-01-28 at 2 17 27 PM

Modal 

The advanced options show only when lookup table format is selected.
Screenshot 2021-01-28 at 2 17 52 PM

Under old CommCare versions, the Edit button is invisible.

## Safety Assurance

- [x] Risk label is set correctly
- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [ ] I am certain that this PR will not introduce a regression for the reasons below

### Automated test coverage

Simple UI change

### QA Plan

Going to have a smoke test, since it's just UI change

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 1]
```diff
@@ -143,15 +160,11 @@ <h4 class="panel-title panel-title-nolink">{% trans "Search Properties" %}</h4>
...
-            <th class="col-sm-4">{% trans "Display Text" %}</th>
-            {% if js_options.default_value_expression_enabled %}
-              <th class="col-sm-3">{% trans "Default Value Expression" %}</th>
-            {% endif %}
-            {% if js_options.search_prompt_appearance_enabled %}
-              <th class="col-sm-2">{% trans "Format" %}</th>
+            <th class="col-sm-4">{% trans "Case Property" %}</th>
+            <th class="col-sm-5">{% trans "Display Text" %}</th>
+            {% if js_options.search_prompt_appearance_enabled or js_options.default_value_expression_enabled%}
+              <th class="col-sm-4">{% trans "More Options" %}</th>
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:166]: Optional: the button doesn't take up that much space, you could give some space back to the property & name text boxes.

**Review by orangejenny — APPROVED** [review]
Nice!

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 2]
```diff
@@ -10,21 +10,21 @@
...
+      <td class="col-sm-4">
         <input class="form-control" type="text" data-bind="value: property.name"/>
       </td>
-      <td class="col-sm-4">
+      <td class="col-sm-6">
         <input class="form-control" type="text" data-bind="value: property.label"/>
       </td>
       {% if js_options.search_prompt_appearance_enabled or js_options.default_value_expression_enabled%}
-      <td class="col-sm-5">
+      <td class="col-sm-4">
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:20]: Something of a nitpick here and below: bootstrap expects the number of columns to add to 12, so this should be 2.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html]: Sure.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html (hunk 2 above)
Comment (sravfeyn) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:20]: Sure

**Review by orangejenny — APPROVED** [review]

**Pull request merged**